src/cmdstan/command.hpp

```diff
@@ -295,29 +211,84 @@ int command(int argc, const char *argv[]) {
+      write_model(diagnostic_csv_writers[i], model.model_name());
+      parser.print(diagnostic_csv_writers[i]);
+    }
   }
 
-  int refresh = get_arg_val<int_argument>(parser, "output", "refresh");
-
-  // arg "init" can be filename or non-negative number (init_radius)
-  std::string init = get_arg_val<string_argument>(parser, "init");
-  double init_radius = 2.0;
-  try {
-    init_radius = boost::lexical_cast<double>(init);
-    init = "";
-  } catch (const boost::bad_lexical_cast &e) {
-  }
-  std::vector<std::shared_ptr<stan::io::var_context>> init_contexts
-      = get_vec_var_context(init, num_chains);
-  int return_code = return_codes::NOT_OK;
-
   //////////////////////////////////////////////////
   //            Invoke Services                   //
   //////////////////////////////////////////////////
-  if (user_method->arg("generate_quantities")) {
+  int return_code = return_codes::NOT_OK;
+  unsigned int id = get_arg_val<int_argument>(parser, "id");
+  int sig_figs = get_arg_val<int_argument>(parser, "output", "sig_figs");
+  int refresh = get_arg_val<int_argument>(parser, "output", "refresh");
+  std::stringstream msg;
+
+  auto user_method = parser.arg("method");
+  if (user_method->arg("pathfinder")) {
+    // ---- pathfinder start ---- //
+    auto pathfinder_arg = parser.arg("method")->arg("pathfinder");
+    int history_size
+        = get_arg_val<int_argument>(*pathfinder_arg, "history_size");
+    double init_alpha
+        = get_arg_val<real_argument>(*pathfinder_arg, "init_alpha");
+    double tol_obj = get_arg_val<real_argument>(*pathfinder_arg, "tol_obj");
+    double tol_rel_obj
+        = get_arg_val<real_argument>(*pathfinder_arg, "tol_rel_obj");
+    double tol_grad = get_arg_val<real_argument>(*pathfinder_arg, "tol_grad");
+    double tol_rel_grad
+        = get_arg_val<real_argument>(*pathfinder_arg, "tol_rel_grad");
+    double tol_param = get_arg_val<real_argument>(*pathfinder_arg, "tol_param");
+    int max_lbfgs_iters
+        = get_arg_val<int_argument>(*pathfinder_arg, "max_lbfgs_iters");
+    bool save_iterations
+        = get_arg_val<bool_argument>(*pathfinder_arg, "save_iterations");
+    int num_elbo_draws
+        = get_arg_val<int_argument>(*pathfinder_arg, "num_elbo_draws");
+    int num_draws = get_arg_val<int_argument>(*pathfinder_arg, "num_draws");
+    int num_psis_draws
+        = get_arg_val<int_argument>(*pathfinder_arg, "num_psis_draws");
+    int num_paths = get_arg_val<int_argument>(*pathfinder_arg, "num_paths");
+    if (num_paths == 1) {
+      return_code = stan::services::pathfinder::pathfinder_lbfgs_single<
+          false, stan::model::model_base>(
+          model, *(init_contexts[0]), random_seed, id, init_radius,
+          history_size, init_alpha, tol_obj, tol_rel_obj, tol_grad,
+          tol_rel_grad, tol_param, max_lbfgs_iters, num_elbo_draws, num_draws,
+          false, refresh, interrupt, logger, init_writer, sample_writers[0],
```

Comment:
Is there a reason why the `save_iterations` argument isn't used (in place of the `false` on this line)?

Reply:
no.  fixed.